Mathlib/Probability/MarcinkiewiczZygmund.lean

```diff
@@ -0,0 +1,187 @@
+/-
+Copyright (c) 2023 Yaël Dillies, Bhavik Mehta. All rights reserved.
+Released under Apache 2.0 license as described ∈ the file LICENSE.
+Authors: Yaël Dillies, Bhavik Mehta
+-/
+import Mathlib.Data.Nat.Choose.Multinomial
+import Mathlib.Probability.IdentDistrib
+
+/-!
+# The Marcinkiewicz-Zygmund inequality
+
+This file proves the Marcinkiewicz-Zygmund inequality.
+
+The Marcinkiewicz-Zygmund inequality states that, if `X₁, ... Xₐ ∈ L^p` are independent random
+variables of mean zero valued in some inner product space, then the `L^p`-norm of `X₁ + ... + Xₐ` is
+at most `Cₚ` times the `L^(p/2)`-norm of `|X₁|² + ... + |Xₐ|²`, where `Cₚ` is a constant depending on
+`p` alone.
+
+## Notation
+
+Throughout this file, `A ^^ n` denotes `A × ... × A` (with `n` factors). Formally, this is
+`Fintype.piFinset fun _ : Fin n ↦ A`.
+
+## TODO
+
+We currently only prove the inequality for `p = 2 * m` an even natural number. The general `p` case
+can be obtained from this specific one by nesting of Lp norms.
+-/
+
+open Finset Fintype Function Nat MeasureTheory ProbabilityTheory Real
+open scoped NNReal
+
+variable {ι Ω E : Type*} {A : Finset ι} {m n : ℕ} [MeasurableSpace Ω] {μ : Measure Ω}
+  [IsFiniteMeasure μ] [mE : MeasurableSpace E] [NormedAddCommGroup E] [InnerProductSpace ℝ E]
+  {X : ι → Ω → E}
+
+local notation:70 A:70 " ^^ " n:71 => Fintype.piFinset fun _ : Fin n ↦ A
+
+/-- The constant appearing in the Marcinkiewicz-Zygmund inequality for symmetric random variables.
+-/
+noncomputable def marcinkiewiczZygmundSymmConst (p : ℝ≥0) : ℝ := (p / 2) ^ (p / 2 : ℝ)
+
+/-- The **Marcinkiewicz-Zygmund inequality** for symmetric random variables, with a slightly better
+constant than `marcinkiewicz_zygmund`. -/
+theorem marcinkiewicz_zygmund_symmetric
+    (iIndepFun_X : iIndepFun (fun _ ↦ mE) X μ)
+    (identDistrib_neg_X : ∀ i, IdentDistrib (X i) (-X i) μ μ)
+    (memℒp_X : ∀ i ∈ A, Memℒp (X i) (2 * m) μ) :
+    ∫ ω, ‖∑ i ∈ A, X i ω‖ ^ (2 * m) ∂μ ≤
+      marcinkiewiczZygmundSymmConst (2 * m) * ∫ ω, (∑ i ∈ A, ‖X i ω‖ ^ 2) ^ m ∂μ := by
+  have : DecidableEq ι := Classical.decEq _
+  -- Turn the `L^p` assumption on the `X i` into various integrability conditions.
+  have integrable_prod_norm_X I (hI : I ∈ A ×ˢ A ^^ m) :
+    Integrable (fun ω ↦ ∏ k, ‖X (I k).1 ω‖ * ‖X (I k).2 ω‖) μ := sorry
+  have integrable_prod_inner_X I (hI : I ∈ A ×ˢ A ^^ m) :
+    Integrable (fun ω ↦ ∏ k, inner (𝕜 := ℝ) (X (I k).1 ω) (X (I k).2 ω)) μ := sorry
```

Comment:
@sgouezel, any idea for how to prove these integrability conditions swiftly? It's hopefully a simple application of Hölder